data-jdbc/src/main/java/io/micronaut/data/jdbc/mapper/ColumnIndexResultSetReader.java

```diff
@@ -106,7 +106,11 @@ public long readLong(ResultSet resultSet, Integer index) {
     @Override
     public char readChar(ResultSet resultSet, Integer index) {
         try {
-            return resultSet.getString(index).charAt(0);
+            String strValue = resultSet.getString(index);
```

Comment:
Should blank string also `''` be checked, not just null, because that could throw an error too?

Reply:
@radovanradic done